lib/openstack_query/query_blocks/query_output.py

```diff
@@ -153,9 +175,51 @@ def generate_output(
         :param openstack_resources: List of openstack objects to obtain properties from - e.g. [Server1, Server2]
         :return: List containing dictionaries of the requested properties obtained from the items
         """
-        return [self._parse_property(item) for item in openstack_resources]
+        output = []
+        for item in openstack_resources:
+            prop_list = self._parse_properties(item)
+            prop_list.update(self._parse_forwarded_outputs(item))
+            output.append(prop_list)
+        return output
 
-    def _parse_property(
+    def _parse_forwarded_outputs(
+        self, openstack_resource: OpenstackResourceObj
+    ) -> Dict[str, str]:
+        """
+        Generates a dictionary of forwarded outputs for the item given
+        :param openstack_resource: openstack resource item to parse forwarded outputs for
+        """
+        forwarded_output_dict = {}
+        if self.forwarded_outputs:
+            for grouped_property, outputs in self.forwarded_outputs.items():
+                prop_val = self._parse_property(grouped_property, openstack_resource)
+
+                # this "should not" error because forwarded outputs should always be a super-set
+                # but sometimes resolving the property fails for whatever reason
+                # in this case just copy all the keys in one of the dictionaries in the list
+                # since they should be equal and set the values to "Not Found"
+                try:
+                    output_list = outputs[prop_val]
+                    # should only ever contain one element since it's grouped by a unique id
+                    forwarded_output_dict.update(output_list[0])
+                except KeyError as exp:
+                    raise QueryChainingError(
+                        "Error: Chaining failed. Could not attach forwarded outputs.\n"
+                        f"Property {grouped_property} extracted from has value {prop_val} which"
+                        "does not match any values from forwarded outputs. "
+                        "This is due to a mismatch in property mappings - likely an Openstack issue"
+                    ) from exp
+
+                    # get keys from first set of forwarded outputs - and set it all to "Not Found"
+                    # (get first item from first group - keys should all be the same)
+                    # TODO: uncomment this block if the error is common
+                    # first_output_entry = list(outputs.values())[0][0]
+                    # forwarded_output_dict.update(
+                    #     {key: "Not Found" for key in list(first_output_entry.keys())}
+                    # )
```

Comment:
Lets put this into an issue / docs and not leave commented code since it tends to rot